core/object/object.cpp

```diff
@@ -1390,9 +1391,6 @@ void Object::disconnect(const StringName &p_signal, const Callable &p_callable)
 bool Object::_disconnect(const StringName &p_signal, const Callable &p_callable, bool p_force) {
 	ERR_FAIL_COND_V_MSG(p_callable.is_null(), false, "Cannot disconnect from '" + p_signal + "': the provided callable is null.");
```

Comment:
Does this line need to be adjusted too? I suspect custom callables will `true` for `is_null()` (which could probably also be changed)

Reply:
I think custom callables will never return `true` from `is_null()`. My understanding is that `is_null` is a completely empty callable (which do to thier nature `CallableCustom`s can never be), `is_valid` is a non-null callable in a state currently callable.

There was a reason I didn't change this line... trying to remember...

Ah, the reason is a callable may become invalid, but it may still be necessary to disconnect it, so allowing non-null but newly invalid callables here satisfies that.

Reply:
> I think custom callables will never return `true` from `is_null()`. My understanding is that `is_null` is a completely empty callable

Yeah, I think that's the desired behavior, but if you look at the implementation of `Callable::is_null()`:

```c++
	_FORCE_INLINE_ bool is_null() const {
		return method == StringName() && object == 0;
	}
```

I think a callable created via `callable_mp_static()` would have an empty `method` and `object`, and so return `true` from `is_null()`. Although, I haven't tested it yet, I'm just reading through the code.

Reply:
Ack! Thanks for pointing that out, I totally didn't notice the `union` :-)

Alright, marking this one as as resolved!